amlip_cpp/include/amlip_cpp/node/collaborative_learning/ModelManagerSenderNode.hpp

```diff
@@ -0,0 +1,170 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file ModelManagerSenderNode.hpp
+ */
+
+#ifndef AMLIPCPP__SRC_CPP_NODE_MODELMANAGERSENDERNODE_HPP
+#define AMLIPCPP__SRC_CPP_NODE_MODELMANAGERSENDERNODE_HPP
+
+#include <functional>
+
+#include <amlip_cpp/node/ParentNode.hpp>
+
+#include <amlip_cpp/types/id/TaskId.hpp>
+#include <amlip_cpp/types/model/ModelDataType.hpp>
+#include <amlip_cpp/types/model/ModelSolutionDataType.hpp>
+#include <amlip_cpp/types/model/ModelStatisticsDataType.hpp>
+
+
+// Forward declaration of dds classes
+namespace eprosima {
+namespace amlip {
+namespace dds {
+
+template <typename T>
+class Writer;
+
+template <typename Data, typename Solution>
+class RPCServer;
+
+} /* namespace dds */
+} /* namespace amlip */
+} /* namespace eprosima */
+
+namespace eprosima {
+namespace amlip {
+namespace node {
+
+/**
+ * @brief Object that listens to:
+ *
+ *  - new ModelDataType messages received from a \c ModelManagerReceiverNode and executes a callback.
+ *
+ * This class is supposed to be implemented by a User and be given to a \c ModelManagerSenderNode in order to process
+ * the messages received from other Nodes in the network.
+ * When data is received, \c send_model is called and it is expected to return a Model Solution for such data.
+ */
+class ModelReplier
+{
+public:
+
+    //! Default virtual dtor so it can be inherited.
+    virtual ~ModelReplier() = default;
+
+    /**
+     * @brief Method that will be called with each ModelSolutionDataType message received to calculate an answer.
+     *
+     * @param status new ModelSolutionDataType message received.
+     */
+    virtual types::ModelSolutionDataType send_model (
+            const types::ModelDataType data) = 0;
+};
+
+/**
+ * @brief This  is a specialisation of the AML-IP node that sends
+ * statistical data from models and receives requests to those models.
+ */
+class ModelManagerSenderNode : public ParentNode
+{
+public:
+
+    /**
+     * @brief Construct a new ModelManagerReceiverNode Node object.
+     *
+     * @param id Id of the Participant (associated with the Node it belongs to)
+     * @param statistics statistical data from models
+     * @param domain_id DomainId of the DDS DomainParticipant
+     */
+    AMLIP_CPP_DllAPI ModelManagerSenderNode(
+            types::AmlipIdDataType id,
+            types::ModelStatisticsDataType& statistics,
```

Comment:
I find it weird that statistics is passed to constructor (unless the attribute was a pointer or reference, which I wouldn't do). In any case, I'd say a setter is missing to update this attribute.